src/acl.rs

```diff
@@ -0,0 +1,189 @@
+                    "|"
+                }
+            };
+
+            if self.can(Permission::Read)   { write!(f, "{}Read",   tick())?; }
+            if self.can(Permission::Write)  { write!(f, "{}Write",  tick())?; }
+            if self.can(Permission::Create) { write!(f, "{}Create", tick())?; }
+            if self.can(Permission::Delete) { write!(f, "{}Delete", tick())?; }
+            if self.can(Permission::Admin)  { write!(f, "{}Admin",  tick())?; }
+            Ok(())
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn permission_bitor() {
+        let all = Permission::Read
+                | Permission::Write
+                | Permission::Create
+                | Permission::Delete
+                | Permission::Admin;
+        assert_eq!(Permission::All, all);
+    }
+
+    #[test]
+    fn permission_can() {
+        assert!(Permission::All.can(Permission::Write));
+        assert!(!Permission::Write.can(Permission::Read));
+    }
+
+    #[test]
+    fn permission_format() {
+        assert_eq!("All", Permission::All.to_string());
+        assert_eq!("None", Permission::None.to_string());
+        assert_eq!("Read|Write", (Permission::Read | Permission::Write).to_string());
+        assert_eq!("Create|Delete", (Permission::Create | Permission::Delete).to_string());
+        assert_eq!("Admin", Permission::Admin.to_string());
+    }
+}
+
+/// An access control list.
+///
+/// In general, the ACL system is similar to UNIX file access permissions, where znodes act as
+/// files. Unlike UNIX, each znode can have any number of ACLs to correspond with the potentially
+/// limitless (and pluggable) authentication schemes. A more surprising difference is that ACLs are
+/// not recursive: If `/path` is only readable by a single user, but `/path/sub` is world-readable,
+/// then anyone will be able to read `/path/sub`.
+///
+/// See the [ZooKeeper Programmer's Guide](https://zookeeper.apache.org/doc/trunk/zookeeperProgrammers.html#sc_ZooKeeperAccessControl)
+/// for more information.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Acl {
+    /// The permissions associated with this ACL.
+    pub perms: Permission,
+    /// The authentication scheme this list is used for. The most common scheme is `"auth"`, which
+    /// allows any authenticated user to do anything (see `creator_all`).
+    pub scheme: String,
+    /// The ID of the user under the `scheme`. For example, with the `"ip"` `scheme`, this is an IP
+    /// address or CIDR netmask.
+    pub id: String,
+}
+
+impl Acl {
+    /// Create a new ACL with the given `permissions`, `scheme`, and `id`.
+    pub fn new<T, U>(permissions: Permission, scheme: T, id: U) -> Acl
+            where T: ToString, U: ToString {
+        Acl {
+            perms: permissions,
+            scheme: scheme.to_string(),
+            id: id.to_string(),
+        }
+    }
+
+    /// This ACL gives the creators authentication id's all permissions.
```

Comment:
The "best practice" of Rust seems to have moved to putting these static-type things into callable functions.